tasks/main.yml

```diff
@@ -117,32 +123,43 @@
     - ansible_distribution in ['CentOS', 'RedHat']
     - ansible_distribution_version is version('9', '>=')
 
-- name: Join to a specific Domain Controller
-  shell: |
-    set -euo pipefail
-    echo {{ ad_integration_password | quote }} | realm join -U \
-      {{ ad_integration_user | quote }} --membership-software \
-      {{ ad_integration_membership_software | quote }} \
-      {{ ad_integration_join_to_dc | quote }}
+- name: Build Command - Join to a specific Domain Controller
+  set_fact:
+    join_command: |
+      set -euo pipefail
+      echo {{ ad_integration_password | quote }} | realm join -U \
```

Comment:
is `ad_integration_password` visible if you do a `ps -ef` or `ps -aux`?

Reply:
@justin-stephenson is there a way to pass the password to `realm` without reading from stdin e.g. by passing a file to read from?

Reply:
This will work but stlil  reads from stdin, other than that there is nothing i'm aware of.
~~~
cat pwfile | realm join ...
~~~

Reply:
This works the same as before, I just decoupled the command preparation and execution.
I would rather not create a file on the disk with password as I consider the related risk higher that password being possibly for a few seconds in the list of processes.